app/components/AreaBooking/index.js

```diff
@@ -17,112 +18,132 @@ import reducer from 'containers/HomePage/reducer';
-import {
-  changeScene,
-  changeSlot,
-  makeReservation,
-} from 'containers/HomePage/actions';
-/* eslint-disable react/prefer-stateless-function */
-
+import { changeScene, makeReservation } from 'containers/HomePage/actions';
+import Calendar from 'components/Calendar';
 import { Wrapper, Div } from './Wrapper';
 
-class AreaBooking extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {};
-  }
-
-  render() {
-    return (
-      <Wrapper>
-        <Div>
-          <LocaleToggle />
-
-          {this.props.scene === 'Loading' && <SceneLoading />}
-          {this.props.scene === 'Setup' && <SceneSetup />}
-          {this.props.scene === 'Start' && (
-            <SceneStart
-              onSubmit={this.props.onChangeSceneToAction}
-              onSelectSlot={this.props.onSelectSlot}
-              freeSlots={this.props.freeSlots}
-              selectedSlot={this.props.selectedSlot}
-            />
-          )}
-          {this.props.scene === 'Action' && (
-            <SceneAction
-              onTimesUp={this.props.onChangeSceneToStart}
-              selectedSlot={this.props.selectedSlot}
-              onButtonClick={this.props.onMakeReservation}
-              onCancelClick={this.props.onChangeSceneToCancel}
-            />
-          )}
-          {this.props.scene === 'Cancel' && (
-            <SceneCancel
-              onTimesUp={this.props.onChangeSceneToStart}
-              onButtonClick={this.props.onChangeSceneToStart}
-            />
-          )}
-          {this.props.scene === 'Verify' && (
-            <SceneVerify
-              onTimesUp={this.props.onChangeSceneToStart}
-              onButtonClick={this.props.onChangeSceneToStart}
-            />
-          )}
-          {this.props.scene === 'Error' && (
-            <SceneError
-              errorMessage={this.props.errorMessage}
-              onButtonClick={this.props.onChangeSceneToStart}
-            />
-          )}
-          {this.props.scene === 'StrongAuth' && (
-            <SceneStrongAuth
-              resource={this.props.resource}
-              resourceId={this.props.resourceId}
-              errorMessage={this.props.errorMessage}
+function useElementSize(ref) {
+  const [size, setSize] = useState([0, 0]);
+
+  useLayoutEffect(() => {
+    function updateSize() {
+      const currentElement = ref.current;
+      if (currentElement) {
+        setSize([ref.current.clientWidth, ref.current.clientHeight]);
+      }
+    }
+
+    window.addEventListener('resize', updateSize);
+    updateSize();
+
+    return () => {
+      window.removeEventListener('resize', updateSize);
+    };
+  }, []);
+
+  return size;
+}
+
+const AreaBooking = ({
+  resource,
+  scene,
+  selectedSlot,
+  onChangeSceneToStart,
+  onMakeReservation,
+  onChangeSceneToCancel,
+  errorMessage,
+  resourceId,
+}) => {
+  const wrapperRef = useRef(null);
+  const [, height] = useElementSize(wrapperRef);
+
```

Comment:
Is the  `,` in front of height needed or just a typo?

Reply:
It's syntax for destructuring an array without declaring a variable for array members I don't have a use for.

So it's the same as `[ width, height]`, but with width is ignored. The purpose is similar to how `_` is used for instance in Haskell or some other fp oriented languages.